lib/Sema/TypeCheckStmt.cpp

```diff
@@ -2139,13 +2139,25 @@ TypeCheckFunctionBodyUntilRequest::evaluate(Evaluator &evaluator,
   if (tc.DebugTimeFunctionBodies || tc.WarnLongFunctionBodies)
     timer.emplace(AFD, tc.DebugTimeFunctionBodies, tc.WarnLongFunctionBodies);
 
+
   tc.validateDecl(AFD);
   tc.checkDefaultArguments(AFD->getParameters(), AFD);
 
   BraceStmt *body = AFD->getBody();
   if (!body || AFD->isBodyTypeChecked())
     return false;
 
+  // If we were asked to skip non-inlinable bodies, and this function is
+  // non-inlinable, then skip typechecking the body. As a conservative step,
+  // still do typecheck functions that appear inside a local context, because
+  // their parent context will have been typechecked.
+  if (tc.SkipNonInlinableFunctionBodies &&
+      !AFD->getDeclContext()->isLocalContext() &&
+      AFD->getResilienceExpansion() == ResilienceExpansion::Maximal) {
+    AFD->setBodyTypecheckingSkipped(true);
+    return false;
```

Comment:
It would be better if we just didn't add these functions to TypeChecker::definedFunctions. We shouldn't get to running this request on them at all